src/services/wallet/providers/metamask.ts

```diff
@@ -0,0 +1,97 @@
+import { ethers } from 'ethers';
+import { ChainConfig, WalletProvider } from '../../../types/wallet';
+
+/**
+ * Implementation of the WalletProvider interface for MetaMask.
+ * Handles all MetaMask-specific wallet operations including connection,
+ * transaction signing, and state management.
+ */
+export class MetamaskWallet implements WalletProvider {
```

Comment:
Down the road, adding additional wallet types would have a similar build where they would `implement WalletProvider` class to ensure required parts of the provider are included for each wallet.